src/pages/index.js

```diff
@@ -118,22 +169,73 @@ function closeModal(modal) {
   modal.removeEventListener("mousedown", handleModalClose);
 }
 
+// api handlers
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error);
 }
 
 function handleCardFormSubmit(evt) {
   evt.preventDefault();
-  const inputValue = { name: cardCaptionInput.value, link: cardUrlInput.value };
-  const cardEl = getCardElement(inputValue);
-  cardsList.prepend(cardEl);
-  closeModal(cardModal);
-  disableButton(cardSubmitButton, settings);
-  evt.target.reset();
+  const submitCardBtn = evt.submitter;
+  submitCardBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: cardCaptionInput.value,
+      link: cardUrlInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardsList.prepend(cardEl);
+      closeModal(cardModal);
+      disableButton(cardSubmitButton, settings);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitCardBtn.textContent = "Save";
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving....";
+  api
+    .editAvatar({
+      avatar: avatarUrlInput.value,
+    })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarEditModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleModalClose(event) {
+  if (event.key === "Escape" && event.type === "keydown") {
+    const openedModalPopUp = document.querySelector(".modal_opened");
+    closeModal(openedModalPopUp);
+  }
+  if (event.target.classList.contains("modal")) {
+    closeModal(event.target);
+  }
 }
+
+//Listeners
 profileButtonEdit.addEventListener("click", () => {
   const formElement = document.querySelector("#edit-modal .modal__form");
```

Comment:
You don't need to search the profile form, because you've already found it as `editFormElement`